go/test/endtoend/reparent/plannedreparent/reparent_test.go

```diff
@@ -557,7 +558,16 @@ func waitForFilePosition(t *testing.T, clusterInstance *cluster.LocalProcessClus
 
 // fileNameFromPosition gets the file name from the position
 func fileNameFromPosition(pos string) string {
-	return pos[0 : len(pos)-4]
+	s := strings.SplitN(pos, ":", 2)
+	if len(s) != 2 {
+		return ""
+	}
+	return s[0]
+}
+
+func TestFileNameFromPosition(t *testing.T) {
```

Comment:
This test is testing a test-helper, which may be overkill. Happy to remove this 👍